neqo-http3/src/frames/hframe.rs

```diff
@@ -78,6 +78,22 @@ impl HFrame {
         }
     }
 
+    /// Is this a known frame type?
+    pub const fn is_known_type(frame_type: HFrameType) -> bool {
+        matches!(
+            frame_type,
+            H3_FRAME_TYPE_DATA
+                | H3_FRAME_TYPE_HEADERS
+                | H3_FRAME_TYPE_CANCEL_PUSH
+                | H3_FRAME_TYPE_SETTINGS
+                | H3_FRAME_TYPE_PUSH_PROMISE
+                | H3_FRAME_TYPE_GOAWAY
+                | H3_FRAME_TYPE_MAX_PUSH_ID
+                | H3_FRAME_TYPE_PRIORITY_UPDATE_REQUEST
+                | H3_FRAME_TYPE_PRIORITY_UPDATE_PUSH
+        )
+    }
+
```

Comment:
How does this differ from:

https://github.com/mozilla/neqo/blob/ca0080e41ec3b06f1813da1c774ca321231e7eec/neqo-http3/src/frames/hframe.rs#L210-L224

Reply:
Related: I find matching non exhaustively on a `u64` (or new type `HFrameType`) error prone, e.g. the compiler does not prevent me from adding a `HFrameType` without updating `is_known_type`. Can one instead match exhaustively on the `enum` `HFrame`?

Reply:
I don't see how, but that might be due to my limited Rust-fu. the `enum` doesn't have any information about what byte value the encoding has.

Reply:
If you are interested in ensuring that there is an exhaustive match, maybe write a function:
```rust
impl TryFrom<u64> for HFrameType {
  fn try_from(v: u64) -> Self {
  }
}
```

Then this could be implemented as:

```rust
fn is_known_type(v: u64) -> bool {
    HFrameType::try_from(v).is_some()
}
```
```

Reply:
`error[E0117]: only traits defined in the current crate can be implemented for primitive types`
when I try

Reply:
You might need to move that to where HFrameType is defined.  Oh, and change it from `type HFrameType = u64` to `struct HFrameType(u64)`.

Reply:
@mxinden even with the latest change, I don't see a way to do what you want. Any pointers?

Reply:
I will take another look. No need to block. I can follow-up with a pull request.

Reply:
I refactored this code locally using an `enum` with [explicit discriminants](https://doc.rust-lang.org/reference/items/enumerations.html#explicit-discriminants):

``` rust
#[derive(Debug, Clone, Copy, PartialEq, Eq)]
#[repr(u64)]
pub enum HFrameTypeKnown {
    Data = 0x0,
    Headers = 0x1,
    CancelPush = 0x3,
    Settings = 0x4,
    PushPromise = 0x5,
    Goaway = 0x7,
    MaxPushId = 0xd,
    PriorityUpdateRequest = 0xf0700,
    PriorityUpdatePush = 0xf0701,
}
```

While it does group the various known types, it complicates the uses outside of `hframe.rs`.

Long story short, I don't think it is worth it. Let's keep it as is. Sorry for the noise.